.github/workflows/workflow.yml

```diff
@@ -178,13 +179,40 @@ jobs:
         echo "${{ secrets.DOCKER_PRIVATE_KEY }}" > ~/.ssh/id
         echo "${{ secrets.DOCKER_KNOWN_HOSTS }}" > ~/.ssh/known_hosts
         ssh linkerd-docker docker version
-    - name: Docker build
+    - name: Build docker images (host)
       if: github.event_name != 'pull_request' || !github.event.pull_request.head.repo.fork
       env:
         DOCKER_HOST: ssh://linkerd-docker
-      run: |
-        export PATH="`pwd`/bin:$PATH"
-        DOCKER_TRACE=1 bin/docker-build
+      run: DOCKER_TRACE=1 bin/docker-build
+    - name: Build docker images (GitHub)
+      if: github.event.pull_request.head.repo.fork
+      run: DOCKER_TRACE=1:LINKERD_LOCAL_BUILD_CLI=1 bin/docker-build
+    - name: Create image archives directory
+      if: github.event.pull_request.head.repo.fork
+      run: mkdir -p $IMAGE_ARCHIVES_DIR
+    - name: Copy out linkerd CLI
+      if: github.event.pull_request.head.repo.fork
+      run: |
+        TAG="$(bin/root-tag)"
+        CLI_IMAGE="gcr.io/linkerd-io/cli-bin:$TAG"
+        CLI_CONTAINER_ID=$(bin/docker create $CLI_IMAGE)
+        bin/docker cp "$CLI_CONTAINER_ID:/out/linkerd-linux" $IMAGE_ARCHIVES_DIR/linkerd
+    - name: Save docker images
+      if: github.event.pull_request.head.repo.fork
+      run: |
+        TAG="$(bin/root-tag)"
+        for img in controller grafana proxy web ; do
+          docker save "gcr.io/linkerd-io/$img:$TAG" > $IMAGE_ARCHIVES_DIR/$img.tar
+        done
+    - name: Move image archives directory for upload action
+      if: github.event.pull_request.head.repo.fork
+      run: mv $IMAGE_ARCHIVES_DIR .
+    - name: Upload image archives
+      if: github.event.pull_request.head.repo.fork
+      uses: actions/upload-artifact@v1
+      with:
+        name: image-archives
+        path: archives
```

Comment:
`path` values do not support environment variables yet ([1](https://github.com/actions/upload-artifact/issues/8)). 

This is a a bit of a hack to get it work well without using `$HOME` since the action runner runs in a subdirectory. I can put a comment saying this can be removed once actions/upload-artifact#8 is closed.

Reply:
Ok I see. If I'm not mistaking the current dir is `$HOME/work/linkerd2/linkerd2`, so a simpler hack (if this allows relative paths) could be to have `path: ../../archives`. WDYT?

Reply:
Sure I can change it to use a relative path in order to get rid of the intermediate `mv`. The ideal solution I would like is for an environment variable to valid here.

That being said I think both approaches (`mv` vs using relative paths) area equally "awkward" solution for the time being. I'll leave a comment either way!